web/frontend/src/JobFootprint.svelte

```diff
@@ -1,232 +1,263 @@
-                mv = round(mean(avgs), 2)
-            } else if (jm?.metric?.series) {
-                mv = round(jm.metric.series[0].statistics.avg, 2)
-            } else {
-                mv = 0.0
-            }
-        }
-        
-        // Unit
-        const fmc = getContext('metrics')(job.cluster, fm)
-        let unit = ''
-        if (fmc?.unit?.base) unit = fmc.unit.prefix + fmc.unit.base
+<script>
+  import { getContext } from "svelte";
+  import {
+    Card,
+    CardHeader,
+    CardTitle,
+    CardBody,
+    Progress,
+    Icon,
+    Tooltip,
+  } from "@sveltestrap/sveltestrap";
+  import { mean, round } from "mathjs";
 
-        // Threshold / -Differences
-        const fmt = findJobThresholds(job, fmc, subclusterConfig)
-        if (fm === 'flops_any') fmt.peak = round((fmt.peak * 0.85), 0)
+  export let job;
+  export let jobMetrics;
+  export let view = "job";
+  export let width = "auto";
 
-        // Define basic data
-        const fmBase = {
-            name: fm,
-            unit: unit,
-            avg: mv,
-            max: fmt.peak
-        }
+  const clusters = getContext("clusters");
+  const subclusterConfig = clusters
+    .find((c) => c.name == job.cluster)
+    .subClusters.find((sc) => sc.name == job.subCluster);
 
-        if (evalFootprint(fm, mv, fmt, 'alert')) {
-            return {
-                ...fmBase,
-                color: 'danger',
-                message:`Metric average way ${fm === 'mem_used' ? 'above' : 'below' } expected normal thresholds.`,
-                impact: 3
-            }
-        } else if (evalFootprint(fm, mv, fmt, 'caution')) {
-            return {
-                ...fmBase,
-                color: 'warning',
-                message: `Metric average ${fm === 'mem_used' ? 'above' : 'below' } expected normal thresholds.`,
-                impact: 2
-            }
-        } else if (evalFootprint(fm, mv, fmt, 'normal')) {
-            return {
-                ...fmBase,
-                color: 'success',
-                message: 'Metric average within expected thresholds.',
-                impact: 1
-            }
-        } else if (evalFootprint(fm, mv, fmt, 'peak')) {
-            return {
-                ...fmBase,
-                color: 'info',
-                message: 'Metric average above expected normal thresholds: Check for artifacts recommended.',
-                impact: 0
-            }
-        } else {
-            return {
-                ...fmBase,
-                color: 'secondary',
-                message: 'Metric average above expected peak threshold: Check for artifacts!',
-                impact: -1
-            }
-        }
-    })
+  const footprintMetrics =
+    job.numAcc !== 0
+      ? job.exclusive !== 1 // GPU
+        ? ["acc_utilization", "acc_mem_used", "nv_sm_clock", "nv_mem_util"] // Shared
+        : ["acc_utilization", "acc_mem_used", "nv_sm_clock", "nv_mem_util"] // Exclusive
+      : (job.exclusive !== 1) // CPU Only
```

Comment:
Adding new hard-coded *required* metrics (without notice in the release notes) is not so great for other users. 🙃

Reply:
I worked around it with the following for now:
```patch
diff --git a/web/frontend/src/JobFootprint.svelte b/web/frontend/src/JobFootprint.svelte
index 8ed8089..62f685e 100644
--- a/web/frontend/src/JobFootprint.svelte
+++ b/web/frontend/src/JobFootprint.svelte
@@ -80,6 +80,18 @@
   const footprintData = footprintMetrics.map((fm) => {
     // Unit
     const fmc = getContext("metrics")(job.cluster, fm);
+    if(fmc === undefined) {
+      return {
+        name: fm,
+        unit: "s",
+        avg: 0,
+        max: 0,
+        color: "info",
+        message: "Metric not configured.",
+        impact: 0,
+      };
+    }
+
     let unit = "";
     if (fmc?.unit?.base) unit = fmc.unit.prefix + fmc.unit.base;
 

```

Reply:
Thanks for the fix. I agree that any hardcoded metric in the backend and the web frontend is bad. We are aware of that, I opened a ticket for this (Issue #260 ) and I will schedule this for the next release.